src/js/components/leftPanel/LeftPanel.tsx

```diff
@@ -0,0 +1,150 @@
+import * as React from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { RootState } from 'js/store/index';
+import { selectActiveTab, toggleTabviewPanel } from 'js/store/appState/actions';
+import ReactTooltip from 'react-tooltip';
+import TabViewContainer from './TabViewContainer';
+import 'css/leftpanel.scss';
+
+import { ReactComponent as LayersTabIcon } from 'images/layersTabIcon.svg';
+import { ReactComponent as AnalysisTabIcon } from 'images/analysisTabIcon.svg';
+import { ReactComponent as DataTabIcon } from 'images/dataTabIcon.svg';
+import { ReactComponent as DocumentsTabIcon } from 'images/documentsTabIcon.svg';
+import { ReactComponent as InfoTabIcon } from 'images/infoTabIcon.svg';
+import { ReactComponent as MeasurementTabIcon } from 'images/MeasurementTabIcon.svg';
+
+export interface TabProps {
+  key: string;
+  label: string;
+  icon: React.SFC<React.SVGProps<SVGSVGElement>>;
+  tooltipText: string;
+  activeTab: string;
+}
+
+// Individual TAB Logic
+const Tab = (props: TabProps): React.ReactElement => {
+  const tabViewVisible = useSelector(
+    (store: RootState) => store.appState.leftPanel.tabViewVisible
+  );
+  const tabIsActive =
+    props.activeTab === props.label && tabViewVisible
+      ? 'tab-button__active'
+      : '';
+  const dispatch = useDispatch();
+  const savedActiveTab = useSelector(
+    (store: RootState) => store.appState.leftPanel.activeTab
+  );
+
+  function handleTabClick(): void {
+    if (savedActiveTab !== props.label) {
```

Comment:
Let's roll all of our `if` statements as _affirmative-first_ if possible --> Switch this statement to show Equality first.
Exceptions exist